sympy/polys/rings.py

```diff
@@ -2531,6 +2531,124 @@ def compose(f, x, a=None):
 
         return poly
 
+
+    def coeff_wrt(self, sym, dg):
+        """
+        Coefficient of ``f`` with respect to ``x**deg``.
+
+        Treating ``f`` as a univariate polynomial in ``x`` this finds the
+        coefficient of ``x**deg`` as a polynomial in the other generators.
+
+        Parameters
+        ==========
+
+        p : sympy.Poly
+            The polynomial to compute the sparse coefficient for.
+        sym : int
+            The symbol to compute the coefficient for.
+        dg : int
+            The degree of the monomial to compute the coefficient for.
+
+        Returns
+        =======
+
+        sympy.Number
+            The sparse coefficient of the polynomial at the given symbol and degree.
+
+        Examples
+        ========
+
+        >>> from sympy.polys import ring, ZZ
+        >>> _, x, y, z = ring("x, y, z", ZZ)
+        >>> p = 2*x**4 + 3*y**4 + 10*z**2 + 3
+        >>> dg = 2
+        >>> sym = 2
+        >>> p.coeff_wrt(sym, dg)
+        10
+
+        """
+        p = self
+        terms = [(m, c) for m, c in p.iterterms() if m[sym] == dg]
+        monoms, coeffs = zip(*terms)
+        monoms = [m[:sym] + (0,) + m[sym + 1:] for m in monoms]
+        return p.ring.from_dict(dict(zip(monoms, coeffs)))
+
+
+    def prem2(self, g, x):
+        """
+        Computes the pseudo-remainder of the polynomial `f` with respect to `g`.
+
+        Parameters
+        ==========
+
+        f : PolyElement
+            The polynomial to compute the pseudo-remainder for.
+        g : PolyElement
+            The polynomial to divide `f` by.
+        x : Symbol
+            The main variable of the polynomials.
+
+        Returns
+        =======
+
+        PolyElement: The pseudo-remainder polynomial.
+
+        Raises
+        ======
+
+        ZeroDivisionError: If the degree of `g` is negative.
+        ValueError: If the algorithm encounters an unexpected condition.
```

Comment:
Saying the "degree is negative" is just a confusing way of saying that `g` is zero. It would be better just to say that directly.

What does "an unexpected condition" mean? We should not list that as something that is raised unless the user can provide inputs that are expected to lead to that exception.

Reply:
Should I use `PolynomialDivisionFailed`instead of `ValueError`?

Reply:
When would the exception be raised?